tests/test_application.py

```diff
@@ -481,7 +481,7 @@ async def test_one_context_per_update(self, app):
         async def one(update, context):
             self.received = context
 
-        def two(update, context):
+        async def two(update, context):
```

Comment:
Why didn't this test fail so far? Does that mean that we have a bug somewhere?

Reply:
No, app.process_error was eating up that error, by logging it instead. I only discovered this accidentally while running pytest with `-rP` which also shows log outputs. Test didn't fail since `await two()` would end up calling the function first and thus the results were expected

Reply:
ah okay. IIRC we configured pytest to convert warnings into exceptions. Maybe we could add a default error handler to the `app` fixture that issues warnings instead of logging? that might help at least a bit with these kinds of things.

Reply:
having multiple error handlers is supported by default though unless one of then raises `ApplicationHandlerStop` 🤔 If introducing something like this makes more problems than it adds benefits, then we should ofc not do that, but maybe it's worth a shot :)

Reply:
tried a couple of solutions, and it seems to be giving more problems so not adding